src/Action/Smtp.hs

```diff
@@ -0,0 +1,123 @@
+    sendMailToAddress :: Address -> EmailMessage -> m ()
+
+    default sendMailToAddress :: MonadIO m => Address -> EmailMessage -> m ()
+    sendMailToAddress = sendMailToAddressIO
+
+sendMailToAddressIO :: (MonadSendMailError e m, MonadReaderConfig r m, MonadIO m) =>
+                       Address -> EmailMessage -> m ()
+sendMailToAddressIO receiver msg = do
+    -- FIXME: when logger gets its config implicitely one can use a "viewSmtpConfig"
+    -- It would be actually even nicer to un-tangle the logger using a MonadWriter.
+    -- Then only the SmtpConfig would be pulled in.
+    cfg <- viewConfig
+    let scfg   = cfg ^. smtpConfig
+        sender = Address (Just $ scfg ^. senderName . to cs) (scfg ^. senderEmail . to cs)
+        mail   = simpleMail' receiver sender (msg ^. msgSubject) (cs $ msg ^. msgBody)
+    r <- liftIO $ do
+        logger cfg {- debug -} $ "sending email: " <> ppShow (receiver, msg)
+        when (isJust $ msg ^. msgHtml) . logger cfg {- warn -} $ "No support for the optional HTML part"
+        renderedMail <- renderMail' mail
+        try $ sendmailCustomCaptureOutput (scfg ^. sendmailPath) (scfg ^. sendmailArgs) renderedMail
+    case r of
+        Right (out, err) -> do
+            liftIO $ do
+                unless (SB.null out) .
+                    logger cfg {- warn -} $ "sendmail produced output on stdout: " <> cs out
+                unless (SB.null err) .
+                    logger cfg {- warn -} $ "sendmail produced output on stderr: " <> cs err
+        Left (e :: IOException) ->
+            throwSendMailError $ IOErrorRunningSendMail e
+  where
+
+sendMailToUser :: HasSendMail e r m => [SendMailFlag] -> User -> EmailMessage -> m ()
+sendMailToUser flags user msg = do
+    case userAddress user of
+        Just address -> sendMailToAddress address msg
+        Nothing
+            | IgnoreMissingEmails `elem` flags -> pure ()
+            | otherwise ->
+                throwSendMailError $ NoEmailAddressForUser user
+
+{-
+newtype EasySendMail = EasySendMail { runEasySendMail :: ReaderT Config (ExceptT SendMailError m))) }
+    deriving (Functor m, Applicative m, Monad m, MonadReader )
+instance (ThrowSendMailError e, GetConfig r, MonadIO m) => HasSendMail e r (EasySendMail e r m)
+-- instance (ThrowSendMailError e, GetConfig r) => HasSendMail e r (
+
+-- | Run sendMail to check that we can send emails. Throw an error if sendmail
+-- is not available or doesn't work.
+checkSendMail :: Config -> IO ()
+checkSendMail cfg = do
+    let address = Address Nothing "user@example.com"
+        msg     = EmailMessage "Test Mail" "This is a test" Nothing
+    r <- runExceptT (runReaderT (sendMailToAddress address msg :: ReaderT Config (ExceptT SendMailError IO) ()) cfg)
+    case r of
+        Left _ -> throwIO $ ErrorCall "sendmail seems to not work.\
+                                    \ Maybe the sendmail path is misconfigured?"
+        Right () -> pure ()
+-}
```

Comment:
Oops, I meant to keep that example/test code alive but I couldn't get rid of the type error quick enough...